servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/MultiAddressCompatibleHostHeaderHttpRequestFilter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2024 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.netty;
+
+import io.servicetalk.context.api.ContextMap;
+import io.servicetalk.http.api.HttpRequestMetaData;
+import io.servicetalk.http.utils.HostHeaderHttpRequesterFilter;
+
+import static io.servicetalk.http.api.HttpHeaderNames.HOST;
+
+final class MultiAddressCompatibleHostHeaderHttpRequestFilter extends HostHeaderHttpRequesterFilter {
+
+    static final ContextMap.Key<CharSequence> AUTHORITY_KEY =
+            ContextMap.Key.newKey("request-authority", CharSequence.class);
```

Comment:
to avoid name clashes in the future, consider following a pattern similar to other key names, i.e. (`MultiAddressCompatibleHostHeaderHttpRequestFilter.class.getName() + ".requestAuthority"`)

I just realized that we have something of everything in terms of style (HTTP_FORCE_NEW_CONNECTION, "share-context-key", "HttpMessageDiscardWatchdogClientFilter.class.getName() + ".messagePublisher") so I guess the dash is fine. We should probably rework all of them to be consistent, but it's low priority.

Reply:
I like the pattern so I'll take it for this case.